testsuite/pytests/mpi/2/test_issue_3099.py

```diff
@@ -0,0 +1,53 @@
+# -*- coding: utf-8 -*-
+#
+# test_issue_3099.py
+#
+# This file is part of NEST.
+#
+# Copyright (C) 2004 The NEST Initiative
+#
+# NEST is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 2 of the License, or
+# (at your option) any later version.
+#
+# NEST is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+
+
+import nest
+import pytest
+
+
+@pytest.fixture
+def conns():
+    nest.ResetKernel()
+    nrn = nest.Create("parrot_neuron")
+    nest.Connect(nrn, nrn)
+    return nest.GetConnections()
+
+
+def test_conn_weight(conns):
+    """Test that operation does cause MPI deadlock."""
```

Comment:
```suggestion
    """Test that operation does not cause MPI deadlock."""
```
Since the test should fail if a deadlock happens, a negation is more correct I think. Also applies to the docstrings below.